src/main/scala/zio/rocksdb/package.scala

```diff
@@ -1,45 +1,88 @@
 package zio
 
 import org.{ rocksdb => jrocks }
-import zio.stream.ZStream
+import zio.stream.{ Stream, ZStream }
 
-package object rocksdb extends RocksDB.Service[RocksDB] {
+package object rocksdb {
   type Bytes = Chunk[Byte]
 
+  type RocksDB = Has[RocksDB.Service]
+
+  object RocksDB {
+    trait Service {
+      def delete(key: Array[Byte]): Task[Unit]
+
+      def delete(cfHandle: jrocks.ColumnFamilyHandle, key: Array[Byte]): Task[Unit]
+
+      def get(key: Array[Byte]): Task[Option[Array[Byte]]]
+
+      def get(cfHandle: jrocks.ColumnFamilyHandle, key: Array[Byte]): Task[Option[Array[Byte]]]
+
+      def multiGetAsList(keys: List[Array[Byte]]): Task[List[Option[Array[Byte]]]]
+
+      def multiGetAsList(keysAndHandles: Map[Array[Byte], jrocks.ColumnFamilyHandle]): Task[List[Option[Array[Byte]]]]
+
+      def newIterator: Stream[Throwable, (Array[Byte], Array[Byte])]
+
+      def newIterator(cfHandle: jrocks.ColumnFamilyHandle): Stream[Throwable, (Array[Byte], Array[Byte])]
+
+      def newIterators(
+        cfHandles: List[jrocks.ColumnFamilyHandle]
+      ): Stream[Throwable, (jrocks.ColumnFamilyHandle, Stream[Throwable, (Array[Byte], Array[Byte])])]
+
+      def put(key: Array[Byte], value: Array[Byte]): Task[Unit]
+
+      def put(cfHandle: jrocks.ColumnFamilyHandle, key: Array[Byte], value: Array[Byte]): Task[Unit]
+    }
+
+    def live(
+      options: jrocks.DBOptions,
+      path: String,
+      cfDescriptors: List[jrocks.ColumnFamilyDescriptor]
+    ): ZLayer.NoDeps[Throwable, Has[(RocksDB.Service, List[jrocks.ColumnFamilyHandle])]] =
```

Comment:
I'm definitely coming at this from the outside a little bit but this signature of returning `Has` with the tuple seems a little strange to me.  Would we want this to instead be `RocksDB with Has[List[jrocks.ColumnFamilyHandle]]`, possibly with a type alias for the second part?

Reply:
Not sure about this one. It looks strange as it's a somewhat special-purpose "factory". @iravid should we try to split, or provide some alternative way to obtain handles?